test/e2e/common/security_context.go

```diff
@@ -69,24 +69,20 @@ var _ = framework.KubeDescribe("Security Context", func() {
 		}
 
 		/*
-		  Release : v1.12
-		  Testname: Security Context: runAsUser (id:65534)
-		  Description: Container created with runAsUser option, passing an id (id:65534) uses that
-		  given id when running the container.
-		  This test is marked LinuxOnly since Windows does not support running as UID / GID.
+			Release : v1.14
+			Testname: Security Context, runAsUser (id:65534)
+			Description: Container is created with runAsUser option by passing an id (id:65534) to run as unpriviledged user. Pod MUST be in Succeeded phase.
 		*/
-		It("should run the container with uid 65534 [LinuxOnly] [NodeConformance]", func() {
+		framework.ConformanceIt("should run the container with uid 65534 [NodeConformance]", func() {
 			createAndWaitUserPod(65534)
 		})
 
 		/*
-		  Release : v1.12
-		  Testname: Security Context: runAsUser (id:0)
-		  Description: Container created with runAsUser option, passing an id (id:0) uses that
-		  given id when running the container.
-		  This test is marked LinuxOnly since Windows does not support running as UID / GID.
+			Release : v1.14
+			Testname: Security Context, runAsUser (id:0)
+			Description: Container is created with runAsUser option by passing an id (id:65534) to run as root priviledged user. Pod MUST be in Succeeded phase.
 		*/
-		It("should run the container with uid 0 [LinuxOnly] [NodeConformance]", func() {
+		framework.ConformanceIt("should run the container with uid 0 [NodeConformance]", func() {
```

Comment:
I don't think that all conformant platforms will allow this.  The test really has to be that *if* the pod is allowed to run, then the UID has to be zero.  It is valid for a platform to not allow uid 0 containers and still be conformant.

Reply:
>It is valid for a platform to not allow uid 0 containers and still be conformant.

If this is the case that **an Conformant platform may not allow to run container's with uid 0** then this e2e can never be part of Conformance suite. Respective non-promotable comment can be added above this e2e.

To make sure "An Conformant platform may not allow to run containers with uid 0" we can tweek **`should run the container with uid 65534 [LinuxOnly] [NodeConformance]`** e2e to use ```
SecurityContext: &v1.SecurityContext{
    runAsNonRoot: true
    RunAsUser: 65534,
},``` option in Pod Spec's SecurityContext. This would verify to 2 behaviors -
1. not allow uid 0 containers
2. run container with uid 65535 (unprivileged user)

Following tweek can be done:
1. Create pod with `runAsNonRoot: true` and log output.(Either may get PASS if container image runs as non-root user by default OR may fail with `CreateContainerConfigError` due to default `root` user in container image)
2. If failed with above error, update Pod to `RunAsUser: 65534,` and Pod MUST pass.

Please let me know it make sense?

Reply:
Please do not remove the ``[LinuxOnly]`` label, test contains a SecurityContext, which is not working on Windows.

Reply:
Not Promoting this e2e to Conformance. Noted reason in the code comment.